### ISSUE TITLE
Auto Import of PRs for a Repository / Account

### ISSUE THREAD

**Issue by keithbro** [apcros/Inspicio#20]
For example, I want all PRs that I create on Github or only for a particular repository to created in Inspicio.

**Comment by apcros** [comment]
That's a good idea. 
I guess you want it to be done in the form of rules you can set on your account ? 
(A list with auto import, and You can tick all the repo you want ?)

**Comment by keithbro** [comment]
Yep that makes sense. Also, when the user signs up you could ask if they want to have everything imported automatically. Personally I think I would want everything imported.

**Comment by apcros** [comment]
Auto-importing at sign-up is going to be tricky one, because you get only 5 points on signup and importing/creating a pull request cost you a point. that mean you'll be able to auto import only 5 open PR, that defeat the purpose of it I think.
(But the auto-import per repo is definitely a good idea, I'll also make sure you get an email if a pr failed to created because you're missing points)

**Comment by keithbro** [comment]
I see.. I had forgotten about the cost of creating a PR. Maybe then the ability to pick up to 5 PRs to import at sign up. Fairly low priority feature, but could be nice.

**Comment by apcros** [comment]
Good point ! I'll have a look to put that in

**Comment by keithbro** [comment]
It would help to get users started and help them understand what the tool is about

**Comment by apcros** [comment]
I decided to split this feature request in two distinct features. 

The first one will be a bulk import feature, to allow you to get at any point a simplified form showing all of your repo with open PR and will allow you to pick as much as you can (depending on your available points)

This is being worked on in this branch : https://github.com/apcros/Inspicio/tree/adding_import

The second will be the auto import, that will allow you to setup rules to import any **new** pull request on any of your repo.

**Comment by apcros** [comment]
The bulk import feature is a lot more work than expected, mainly because I've made some poor structural choices. 
![Poor choices](http://www.reactiongifs.us/wp-content/uploads/2013/06/raining_david_tennant.gif)

This is still being work on.

Still need : 
- Bitbucket pull request information fetching
- Front end results when bulk complete is finished
- Way more tests

Once this is done, I might do #28 (https://laravel.com/docs/5.4/eloquent ) before the Auto import, as this is getting more and more complicated to validate everything I'm doing without proper DB models.

**Comment by apcros** [comment]
Wooo bulk-import is now in review https://inspicio.tech/reviews/8e096eb0-3d9f-462c-b89f-30de4b5cdf12/view with not as much tests as I would've liked, mainly because of how I've fucked the dependency injection.
This highly the urgency of #28 to be completed in order to add more tests to everything relying on HTTP requests.
(I wish mocking was as easy in PHPunit/Laravel as it is with Test::Most/Mojo )

Anyway, let me know what you think, once this is okay I'll merge and deploy it and tackle the refactor mentionned above. (And then the Auto import)

**Comment by apcros** [comment]
Bulk import has now been merged and deployed.
As said above I'll now work on #28 before continuing on the second part of your request.

**Comment by apcros** [comment]
The auto-import is (finally) being worked on. Here's the branch : https://github.com/apcros/Inspicio/tree/auto_import

**Comment by apcros** [comment]
Auto import is finally in code review ! https://github.com/apcros/Inspicio/pull/34

**Comment by apcros** [comment]
This has now been released and it's ready to be used :)
Feel free to re-open the issue @keithbro if you're unhappy with how this has been implemented
